data/src/main/java/io/gnosis/data/repositories/TransactionRepository.kt

```diff
@@ -0,0 +1,180 @@
+        val value = transaction.dataDecoded?.parameters?.getValueByName("value")?.decimalAsBigInteger() ?: BigInteger.ZERO
+
+        //Only available with transferFrom
+        val from = transaction.dataDecoded?.parameters?.getValueByName("from")?.asEthereumAddress()
+
+        return Transaction.Transfer(
+            to,
+            from ?: transaction.safe,
+            value,
+            transaction.executionDate?.formatBackendDate()
+                ?: transaction.submissionDate?.formatBackendDate()
+                ?: transaction.modified?.formatBackendDate(),
+            TokenRepository.FAKE_ERC20_TOKEN_INFO // TODO: find out correct token data source
+        )
+    }
+
+    private fun transferErc721(transaction: MultisigTransactionDto): Transaction.Transfer =
+        Transaction.Transfer(
+            transaction.to,
+            transaction.safe,
+            transaction.value,
+            transaction.executionDate?.formatBackendDate(),
+            TokenRepository.FAKE_ERC721_TOKEN_INFO // TODO: find out correct token data source
+        )
+
+    private fun settings(transaction: MultisigTransactionDto): Transaction.SettingsChange =
+        Transaction.SettingsChange(
+            transaction.dataDecoded!!,
+            transaction.executionDate?.formatBackendDate(),
+            transaction.nonce
+        )
+
+    private fun custom(transaction: ModuleTransactionDto): Transaction.Custom =
+        Transaction.Custom(
+            transaction.nonce,
+            transaction.module,
+            transaction.data?.dataSizeBytes() ?: 0L,
+            transaction.created,
+            transaction.value ?: BigInteger.ZERO
+        )
+
+    private fun custom(transaction: MultisigTransactionDto): Transaction.Custom =
+        Transaction.Custom(
+            transaction.nonce,
+            transaction.to,
+            transaction.data?.dataSizeBytes() ?: 0L,
+            transaction.creationDate,
+            transaction.value
+        )
+
+    private fun custom(transaction: EthereumTransactionDto): Transaction.Custom =
+        Transaction.Custom(
+            null, // Ethereum txs do not have a nonce
+            transaction.from,
+            transaction.data?.dataSizeBytes() ?: 0L,
+            transaction.blockTimestamp,
+            transaction.value ?: BigInteger.ZERO
+        )
+
+    private fun custom(transaction: TransactionDto): Transaction.Custom =
+        Transaction.Custom(
+            null,
+            transaction.to,
+            transaction.data?.dataSizeBytes() ?: 0L,
+            null,
+            BigInteger.ZERO
+        )
+
+}
+
+fun List<ParamsDto>?.getValueByName(name: String): String? {
+    this?.map {
+        if (it.name == name) {
+            return it.value
+        }
+    }
+    return null
+}
+
+fun String.dataSizeBytes(): Long = removeHexPrefix().hexToByteArray().size.toLong()
+fun String?.hexStringNullOrEmpty(): Boolean = this?.dataSizeBytes() ?: 0L == 0L
```

Comment:
I think that we must have a method for this in svalinn. Also, this check could be simplified to `this?.size > 2 == true`, where two accounts for `0x` but definitely have a look `:utils` in svalinn.

Reply:
I did not find a function (String? -> Boolean) in svalin:util. Also ">" doesn't work because of the nullability. I don't think there is a simpler way to write this. There is however a more readable way: 
```
fun String?.hexStringNullOrEmpty(): Boolean = when {
    this == null -> true
    length > 2 -> false
    else -> true
}
```

Reply:
sure, you are right, nullability is not addressed. I would suggest then `his?.size ?: 0 > 2 == true` but this is really unimportant.